src/request.rs

```diff
@@ -87,7 +87,9 @@ pub struct Request {
     pub is_supported: bool,
     pub is_third_party: bool,
     pub url: String,
+    pub url_lower_cased: String,
     pub hostname: String,
+    pub request_tokens: Vec<utils::Hash>,
```

Comment:
let's make these new fields `pub(crate)`, they're an internal implementation detail and we don't want them to be set outside of the constructors